src/features/unified-share-modal/SharedLinkSection.js

```diff
@@ -335,6 +352,17 @@ class SharedLinkSection extends React.Component<Props, State> {
                         )}
                     </div>
                 )}
+                {accessLevel === ANYONE_IN_COMPANY && isPermissionElevatedToEdit && (
```

Comment:
This warning copy is only for the special case where accessLevel is `ANYONE_IN_COMPANY` and when the permissionLevel has been elevated to Edit from another permission level.